src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -2044,7 +2081,23 @@ private async ValueTask<JsonRpcMessage> DispatchIncomingRequestAsync(JsonRpcRequ
                     }
                 }
 
-                return await this.DispatchRequestAsync(request, targetMethod, cancellationToken).ConfigureAwait(false);
+                request.TryGetTopLevelProperty<string>(JoinableTaskTokenHeaderName, out string? parentToken);
+                if (this.JoinableTaskFactory is null)
+                {
+                    this.joinableTaskTokenWithoutJtf.Value = parentToken;
+                }
+
+                if (this.JoinableTaskFactory is null || parentToken is null)
```

Comment:
If there is a JTF instance but no token was provided in the header, should we still use JTF to dispatch for the benefit of the current process? (although I am not sure what benefit there would be)

Reply:
Lifeng and I couldn't think of any benefit in doing so. But there _is_ a non-zero overhead from doing so. Thus, we only do it when we feel it's justified by skipping when there is no parent token.